src/fileActions.jsx

```diff
@@ -563,3 +526,137 @@ const setDirectoryName = (val, setName, setNameError, setErrorMessage) => {
         setNameError(null);
     }
 };
+
+export const fileActions = (path, files, selected, setSelected, clipboard, setClipboard, addAlert, Dialogs) => {
+    const currentPath = path.join("/") + "/";
+    const directory_name = path[path.length - 1];
+    const menuItems = [];
+    console.log(path, currentPath, directory_name, selected);
+
+    const createLink = (currentPath, files, selected) => {
+        Dialogs.show(
+            <CreateLinkModal
+              currentPath={currentPath} selected={selected}
+              files={files.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase())
+                  ? 1
+                  : ((b.name.toLowerCase() > a.name.toLowerCase())
+                      ? -1
+                      : 0))}
+            />
+        );
+    };
+
+    if (selected.length === 0 || selected[0].name === path[path.length - 1]) {
+        menuItems.push(
+            {
+                id: "paste-item",
+                title: _("Paste"),
+                onClick: () => spawnPaste(clipboard, currentPath, false, addAlert),
+                isDisabled: clipboard === undefined
+            },
+            {
+                id: "paste-as-symlink",
+                title: _("Paste as symlink"),
+                onClick: () => spawnPaste(clipboard, currentPath, true, addAlert),
+                isDisabled: clipboard === undefined
+            },
+            { type: "divider" },
+            {
+                id: "create-item",
+                title: _("Create directory"),
+                onClick: () => Dialogs.show(<CreateDirectoryModal currentPath={currentPath} />),
+            },
+            {
+                id: "create-link",
+                title: _("Create link"),
+                onClick: () => createLink(currentPath, files, selected || {})
+            },
+            { type: "divider" },
+            {
+                id: "edit-permissions",
+                title: _("Edit permissions"),
+                onClick: () => editPermissions(Dialogs, selected[0], path)
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-307-20240307-084539-34707c5b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)